packages/protocol/contracts/layer1/shasta/impl/Inbox.sol

```diff
@@ -563,34 +562,29 @@ abstract contract Inbox is EssentialContract, IInbox {
         );
     }
 
-    /// @dev Attempts to process a forced inclusion from the ForcedInclusionStore
-    /// @notice Uses low-level call to handle potential failures gracefully
+    /// @dev Processes multiple forced inclusions from the ForcedInclusionStore
+    /// @notice Consumes up to _numForcedInclusions from the queue and proposes them sequentially
     /// @param _config Configuration containing forced inclusion store address
-    /// @param _coreState Current core state to update if inclusion processed
-    /// @return coreState_ Updated state if forced inclusion processed, unchanged otherwise
-    /// @return forcedInclusionProcessed_ True if a forced inclusion was successfully processed
-    function _processForcedInclusion(
+    /// @param _coreState Current core state to update with each inclusion processed
+    /// @param _numForcedInclusions Maximum number of forced inclusions to process
+    /// @return Updated core state after processing all consumed forced inclusions
+    function _processForcedInclusions(
```

Comment:
we probably want to return the actual number of forced inclusion processed, then use this number to check if there are still room for the regular proposal.